numba_dppy/tests/test_debuginfo.py

```diff
@@ -142,3 +142,83 @@ def foo(arr):
         for tag in ir_tags:
             got = make_check(kernel_ir, tag)
             assert expect == got
+
+
+def test_debug_flag_generates_ir_with_debuginfo_for_func(offload_device):
+    """
+    Check debug info is emitting to IR if debug parameter is set to True
+    """
+
+    if skip_test(offload_device):
+        pytest.skip()
+
+    if offload_device in "level_zero:gpu:0":
+        pytest.xfail("Failing compilation: SyclProgramCompilationError")
+
+    @dppy.func(debug=True)
+    def func_sum(a, b):
+        result = a + b
+        return result
+
+    @dppy.kernel(debug=True)
+    def data_parallel_sum(a, b, c):
+        i = dppy.get_global_id(0)
+        c[i] = func_sum(a[i], b[i])
+
+    ir_tag_func_sum = r'\!DISubprogram\(name: "test_debug_flag_generates_ir_with_debuginfo_for_func.<locals>.func_sum"'
+    ir_tag_data_parallel_sum = r'\!DISubprogram\(name: "test_debug_flag_generates_ir_with_debuginfo_for_func.<locals>.data_parallel_sum"'
```

Comment:
Regular expression depends on test name. Imagine somebody renames the test and the test fails.

Reply:
We have no way to test correctly without the full name

Reply:
Why? This is regular expression. Just use `.*` instead of test name.